ebean-core/src/main/java/io/ebeaninternal/server/core/DefaultServer.java

```diff
@@ -586,11 +586,7 @@ public void clearQueryStatistics() {
    */
   @Override
   public <T> T createEntityBean(Class<T> type) {
-    final BeanDescriptor<T> desc = descriptor(type);
-    if (desc == null) {
-      throw new IllegalArgumentException("No bean type " + type.getName() + " registered");
-    }
-    return desc.createBean();
+    return descriptor(type).createBean();
```

Comment:
Code change: I moved this check to `descriptor(class)` as it claims to never return null due `@NonNullApi`

Reply:
Hmm. Unfortunaltely, this is not completely true.
While DefaultServer is annotated with `@NonNullApi`, `SpiEbeanServer` does not. And we have also some code, in our app, where we check if a class is registered. So it  might be better to annotate descriptor with @Nullable